plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/connection/OAuthDetailView.java

```diff
@@ -320,33 +321,39 @@ public IAuthorizationDetails getDetails() {
 		@Override
 		protected IStatus run(IProgressMonitor monitor) {
 			try {
-				IClient client = new ClientFactory().create(host, OpenShiftCoreUIIntegration.getInstance().getSSLCertificateCallback());
+				IClient client = new ClientBuilder(host)
+						.sslCertificateCallback(OpenShiftCoreUIIntegration.getInstance().getSSLCertificateCallback())
+						.build();
 				details = client.getAuthorizationDetails(host);
+				return ValidationStatus.OK_STATUS;
 			} catch(Exception e) {
 				if (e.getCause() instanceof ConnectTimeoutException) {
-					return new Status(IStatus.ERROR, OpenShiftUIActivator.PLUGIN_ID, "Timed out waiting for a response for authorization details.\nThis server might be unavailable or may not support OAuth.", e);
+					return new Status(IStatus.ERROR, OpenShiftUIActivator.PLUGIN_ID, 
+							"Timed out waiting for a response for authorization details.\nThis server might be unavailable or may not support OAuth.", e);
+				} else {
+					return new Status(IStatus.ERROR, OpenShiftUIActivator.PLUGIN_ID, 
+						"Unable to retrieve the authentication details", e);
 				}
-				return new Status(IStatus.ERROR, OpenShiftUIActivator.PLUGIN_ID, "Unable to retrieve the authentication details", e);
 			}
-			return ValidationStatus.OK_STATUS;
 		}
 		
 	}
 	
-	private class BearTokenAuthenticationProvider implements IConnectionAuthenticationProvider{
+	private class BearTokenAuthenticationProvider implements IConnectionAuthenticationProvider {
 
 		@Override
 		public IConnection update(IConnection conn) {
 			Assert.isLegal(conn instanceof Connection);
+
 			final Connection connection = (Connection) conn;
 			// might be called from job, switch to display thread to access observables
 			Display.getDefault().syncExec(new Runnable() {
 				
 				@Override
 				public void run() {
 					connection.setAuthScheme(IAuthorizationContext.AUTHSCHEME_OAUTH);
-					connection.setToken((String)tokenObservable.getValue());
-					connection.setRememberToken((boolean)rememberTokenObservable.getValue());
+					connection.setToken((String) tokenObservable.getValue());
+					connection.setRememberToken((boolean) rememberTokenObservable.getValue());
```

Comment:
Since definition of rememberTokenObservable and tokenObservable now has type parameter, the casting may be not needed at all.